packages/expect-puppeteer/src/matchers/toClick.js

```diff
@@ -1,8 +1,11 @@
 import toMatchElement from './toMatchElement'
+import { getContext } from '../utils'
 
 async function toClick(instance, selector, options) {
+  const { page } = await getContext(instance, () => document)
   const element = await toMatchElement(instance, selector, options)
-  await element.click(options)
+
+  await page.evaluate((el, opts) => el.click(opts), element, options)
```

Comment:
This won't work to add the options back.

This `el.click()` is running the [`click`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click) method in javascript not puppeteers [`elementHandle.click`](https://pptr.dev/#?product=Puppeteer&version=v9.0.0&show=api-elementhandleclickoptions) method. So those options would be ignored.

In order to add the options back we would have to duplicate the logic of puppeteer's `elementHandle.click`, which is why I think this would be better to fix this in puppeteer so we don't have to duplicate their logic.